### PR TITLE
[neuralnet] Iteration bug fix in learning rate

### PR DESCRIPTION
**Pull request by kparichay**
learning rate is decayed using the iteration
however current implementation was using epoch count

Note: iteration is saved so that it can be saved with optimizer to continue training if needed.

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: nntrainer/src/neuralnet.cpp [hunk 1]
```diff
@@ -91,19 +91,7 @@ std::vector<std::string> parseLayerName(std::string ll) {
...
-  decay_steps = 0.0;
-  epoch = 0;
-  loss = 0.0;
-  cost = COST_UNKNOWN;
-  weight_ini = WEIGHT_UNKNOWN;
-  net_type = NET_UNKNOWN;
-  data_buffer = NULL;
-  config = "";
-}
+NeuralNetwork::NeuralNetwork() : NeuralNetwork("") {}
```
Comment (jijoongmoon) [inline nntrainer/src/neuralnet.cpp:94]: These variables should be initialized. Otherwise,  "Analysishub service" will give us error.

Comment (jijoongmoon) [inline nntrainer/src/neuralnet.cpp:94]: nice catch!

File: nntrainer/src/neuralnet.cpp [hunk 2]
```diff
@@ -792,15 +780,14 @@ int NeuralNetwork::train_run() {
       vec_4d in, label;
       if (data_buffer->getDataFromBuffer(nntrainer::BUF_TRAIN, in, label)) {
         status =
-          backwarding(nntrainer::Tensor(in), nntrainer::Tensor(label), i);
+          backwarding(nntrainer::Tensor(in), nntrainer::Tensor(label), iter++);
         if (status != ML_ERROR_NONE) {
```
Comment (jijoongmoon) [inline nntrainer/src/neuralnet.cpp:784]: Right. should use saved value.!
